drivers/sensor/shtcx/shtcx.c

```diff
@@ -0,0 +1,273 @@
+#include <drivers/sensor.h>
+#include <sys/__assert.h>
+#include <logging/log.h>
+
+#include "shtcx.h"
+
+LOG_MODULE_REGISTER(SHTCX, CONFIG_SENSOR_LOG_LEVEL);
+
+/* all cmds read temp first: cmd[MEASURE_MODE][Clock_stretching_enabled] */
+static const uint16_t measure_cmd[2][2] = {
+	{ 0x7866, 0x7CA2 },
+	{ 0x609C, 0x6458 },
+};
+
+/* measure_wait_us[shtcx_chip][MEASURE_MODE] */
+static const uint16_t measure_wait_us[2][2] = {
+	/* shtc3: 12.1ms, 0.8ms */
+	{ 1210, 800 }, /* shtc3 */
+	/* shtc1: 14.4ms, 0.94ms */
+	{ 14400, 940 }, /* shtc1 */
+};
+
+/*
+ * CRC algorithm parameters were taken from the
+ * "Checksum Calculation" section of the datasheet.
+ */
+static uint8_t shtcx_compute_crc(uint16_t value)
+{
+	uint8_t buf[2] = { value >> 8, value & 0xFF };
+	uint8_t crc = 0xFF;
+	uint8_t polynom = 0x31;
+	int i, j;
+
+	for (i = 0; i < 2; ++i) {
+		crc = crc ^ buf[i];
+		for (j = 0; j < 8; ++j) {
+			if (crc & 0x80) {
+				crc = (crc << 1) ^ polynom;
+			} else {
+				crc = crc << 1;
+			}
+		}
+	}
+
+	return crc;
+}
+
+/* val = -45 + 175 * sample / (2^16) */
+static void shtcx_temperature_from_raw(uint16_t raw, struct sensor_value *val)
+{
+	int32_t tmp;
+
+	tmp = (int32_t)raw * 175U - (45 << 16);
+	val->val1 = tmp / 0x10000;
+	/* x * 1.000.000 / 65.536 == x * 15625 / 2^10 */
+	val->val2 = ((tmp % 0x10000) * 15625U) / 1024;
+}
+
+/* val = 100 * sample / (2^16) */
+static void shtcx_humidity_from_raw(uint16_t raw, struct sensor_value *val)
+{
+	uint32_t tmp;
+
+	tmp = (uint32_t)raw * 100U;
+	val->val1 = tmp / 0x10000;
+	/* x * 1.000.000 / 65.536 == x * 15625 / 1024 */
+	val->val2 = (tmp % 0x10000) * 15625U / 1024;
+}
+
+static int shtcx_write_command(const struct device *dev, uint16_t cmd)
+{
+	uint8_t tx_buf[2];
+
+	sys_put_be16(cmd, &tx_buf[0]);
+	return i2c_write(shtcx_i2c_bus(dev), tx_buf, sizeof(tx_buf),
+			 shtcx_i2c_address(dev));
+}
+
+static int shtcx_read_words(const struct device *dev, uint16_t cmd, uint16_t *data,
+		     uint16_t num_words, uint16_t max_duration_us)
+{
+	const struct shtcx_config *cfg = dev->config;
+	int status = 0;
+	uint32_t raw_len = num_words * (SHTCX_WORD_LEN + SHTCX_CRC8_LEN);
+	uint16_t temp16;
+	uint8_t rx_buf[SHTCX_MAX_READ_LEN];
+	int dst = 0;
+
+	status = shtcx_write_command(dev, cmd);
+	if (status != 0) {
+		LOG_DBG("Failed to initiate read");
+		return -EIO;
+	}
+
+	if (!cfg->clock_stretching) {
+		k_sleep(K_USEC(max_duration_us));
+	}
+
+	status = i2c_read(shtcx_i2c_bus(dev), rx_buf, raw_len,
+			  shtcx_i2c_address(dev));
+	if (status != 0) {
+		LOG_DBG("Failed to read data");
+		return -EIO;
+	}
+
+	for (int i = 0; i < raw_len; i += (SHTCX_WORD_LEN + SHTCX_CRC8_LEN)) {
+		temp16 = sys_get_be16(&rx_buf[i]);
+		if (shtcx_compute_crc(temp16) != rx_buf[i+2]) {
+			LOG_DBG("invalid received invalid crc");
+			return -EIO;
+		}
+
+		data[dst++] = temp16;
+	}
+
+	return 0;
+}
+
+static int shtcx_sleep(const struct device *dev)
+{
+	if (shtcx_write_command(dev, SHTCX_CMD_SLEEP) < 0) {
+		return -EIO;
+	}
```

Comment:
Add empty line between } and return line in the functions above.
Same for L:145, L:248

Reply:
done